doc/iterations/2025.4/encode_workbench_state_in_url.adoc

```diff
@@ -0,0 +1,109 @@
+
+== Problem
+
+When opening a project, the URL always starts with: `/projects/:projectId/edit/`.
+
+If at least 1 representation editor is opened, then the URL starts with: `/projects/:projectId/edit/:representationId` where `representationId` is the ID of the representation opened in the active editor.
+
+And since https://github.com/eclipse-sirius/sirius-web/issues/4376[#4376], the URL also contains the current selection of the workbench, meaning the URL can be of the form `/projects/:projectId/edit/?selection=id1%2Cid2` or `/projects/:projectId/edit/:representationId?selection=id1%2Cid2` (where `%2C` encodes `,`).
+
+Reversely, when resolving one of the above-mentioned URLs, the corresponding project, representation and selections are set.
+
+However, there are parts of the workbench state which are not encoded in the URL.
+
+These parts are the views opened in the left and right panels (e.g. in Sirius Web application, between `Explorer` and `Validation` on the left, and between `Details`, `Query`, `Representations` and `Related Elements` on the right).
+When loading a URL, the default views (`Explorer` and `Details`) are opened instead.
+Moreover, these views may be configurable (e.g. in view `Explorer`, the tree representation may be changed, and filters may be applied).
+As for editors, since only the active representation editor gets encoded into the URL (through the ID of the opened representation), other opened editors are not opened upon resolution of the URL.
+
+Being able to share the complete workbench state through the application URL would facilitate collaboration between users.
+
+== Key Result
+
+The URL should be synchronized with the full state of the workbench.
+
+=== Acceptance Criteria
+
+* The URL is synchronized with the full state of the workbench, which includes:
+** The active view on the left (and its configuration)
+** The opened editors in the center
+** The active view on the right (and its configuration)
+** The focused part of the workbench (one of the active views or the active editor)
+** The selection
+* When resolving a URL that specifies a workbench state, the matching workbench state is set up.
+
+== Solution
+
+* We will rework the URL to change based on the state of the workbench.
+* The base URL for an open project will be `/projects/:projectId/edit`.
+* The state of the workbench will be encoded in the URL through the following parameters:
+** `views`: will encode which views are active, e.g. `explorer%2Cdetails`.
+** `editors`: will encode which editors are opened, through the ID of their input, e.g. `representationId1%2CrepresentationId2%2CrepresentationId3`. (Note: what's the real difference between views and editors in this context? Maybe we should merge views and editors as parts).
+** `configurations`: will encode the configurations of the active views and editors, e.g. `explorer_state_closed%2Cexplorer_tree_activeFilterId_filterId1`.
```

Comment:
It's a bit odd to store the configuration in another property, you could technically encode a configuration for a view which is not opened